platform/src/main/java/org/hillview/maps/LinearProjectionMap.java

```diff
@@ -0,0 +1,72 @@
+package org.hillview.maps;
+
+import org.hillview.dataset.api.IMap;
+import org.hillview.table.*;
+import org.hillview.table.api.ContentsKind;
+import org.hillview.table.api.IColumn;
+import org.hillview.table.api.IRowIterator;
+import org.hillview.table.api.ITable;
+import org.jblas.DoubleMatrix;
+import org.jblas.ranges.AllRange;
+import org.jblas.ranges.PointRange;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This map takes a list of column names and a projection matrix,
+ * and applies the projection matrix to the matrix that is
+ * constructed from the table by horizontally stacking the
+ * specified columns. The resulting table is a copy of the old
+ * table, with the additional projected columns added to it.
+ */
+public class LinearProjectionMap implements IMap<ITable, ITable> {
+
+    private final DoubleMatrix projectionMatrix;
+    private final String[] colNames;
```

Comment:
You should probably document the projection matrix - in particular, the order of columns has to be the same as the colNames array.